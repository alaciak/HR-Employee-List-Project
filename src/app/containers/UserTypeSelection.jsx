import React from 'react'
import { connect } from 'react-redux';
import { selectUserType } from '../actions/userTypeSelectionActions';

export class UserTypeSelection extends React.Component {

  handleOnChangeOption = e => {
    this.props.selectUserType(e.target.value);
  }

  render() {
    return (
      <select className='navigation-user_select' name='loggedIn' value={ this.props.userType } onChange={ this.handleOnChangeOption }>
        <option value='user'>User</option>
        <option value='admin'>Admin</option>
      </select>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    userType: state.userTypeSelectionReducer.userType
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    selectUserType: (userSelected) => {
      dispatch(selectUserType(userSelected));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserTypeSelection);
