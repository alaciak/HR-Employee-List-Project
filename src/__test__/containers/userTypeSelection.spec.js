import React from 'react';
import { UserTypeSelection } from '../../app/containers/UserTypeSelection.jsx';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('UserTypeSelection', () => {

    const testUser = 'admin';

    it('should render if the userType is passed', () => {

    const wrapper = shallow(<UserTypeSelection userType={ testUser }/>);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
