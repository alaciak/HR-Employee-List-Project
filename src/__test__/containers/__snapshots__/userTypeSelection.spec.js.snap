// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTypeSelection should render if the userType is passed 1`] = `
<select
  className="navigation-user_select"
  name="loggedIn"
  onChange={[Function]}
  value="admin"
>
  <option
    value="user"
  >
    User
  </option>
  <option
    value="admin"
  >
    Admin
  </option>
</select>
`;
